File: lab0 (Pascal)/lab0.hs
{-# LANGUAGE OverloadedStrings #-}

import Control.Monad
import Network.HTTP.Conduit
import Data.Text.Encoding
import qualified Data.ByteString.Lazy as L
import qualified Data.ByteString.Char8 as C
import Network (withSocketsDo)

(email, name) = ("sautin@phystech.edu", encodeUtf8 "Саутин А.А. (2)") -- адрес почты и фамилия с инициалами

pascal :: Int -> Int -> Int
pascal c r
  | c < 0 || c > r || r < 0 = 0
  | r == c || c == 0 = 1
  | otherwise = (pascal c (r-1)) + (pascal (c-1) (r-1))

printIt :: Int -> C.ByteString
printIt n = C.pack $ show $ [pascal y x | x <- [0..n], y <- [0..x]]

main :: IO()
main = 
  withSocketsDo $ do
  initReq <- parseUrl "http://mipt.eu01.aws.af.cm/lab0"
  let req = urlEncodedBody [("email", email), ("name", name), ("lang", "haskell"), ("content", printIt 20)] $ initReq { method = "POST" }
  response <- withManager $ httpLbs req
  L.putStr $ responseBody response

File: lab0 (Pascal)/lab0.fs
module lab0

open System
open System.Net
open System.Collections.Specialized

let (email, name) = ("sautin1@yandex.ru", "Саутин А.А.") // адрес почты и фамилия с инициалами

let rec pascal c r = 
  match (c,r) with // а тут решение
    | _ when c < 0 -> 0
    | _ when r < 0 -> 0
    | _ when c > r -> 0
    | _ when r = c -> 1
    | (0,r) -> 1
    | (c,r) -> (pascal c (r-1)) + (pascal (c-1) (r-1))

let printIt n = 
  "[" +
  ([for x in 0..n do for y in 0..x do yield pascal y x] 
    |> List.map (fun x -> x.ToString())
    |> List.reduce (fun x y -> x + "," + y) )
  + "]"

let main () = 
  let values = new NameValueCollection()
  values.Add("email", email)
  values.Add("name", name)
  values.Add("lang", "fsharp")
  values.Add("content", printIt 20)

  let client = new WebClient()
  let response = client.UploadValues(new Uri("http://mipt.eu01.aws.af.cm/lab0"), values)
  let responseString = Text.Encoding.Default.GetString(response)

  printf "%A\n" responseString
